### ISSUE TITLE
XML syntax highlighter aborts prematurely

### ISSUE THREAD

**Issue by skycommand** [Microsoft/vscode#47375]
Issue Type: <b>Bug</b>

The XML syntax highlighter of Visual Studio Code refuses to properly highlight any chunk of XML code that (1) has no line breaks and (2) is larger than exactly 20,000 characters. Most XML files that I edit, however, have no line breaks at all.

[I have uploaded a `test.xml` file](https://github.com/Microsoft/vscode/files/1886059/Test.zip) which can be used to reproduce the issue. To reproduce:

1. Open `test.xml` in Visual Studio Code.
2. Observe that there is an unformatted chunk in it.
3. Delete on character from anywhere in this chunk or insert a link break anywhere in it.

You will observe that after step 3, the non-highlighted chunk is suddenly highlighted.

VS Code version: Code 1.22.1 (950b8b0d37a9b7061b6f0d291837ccc4015f5ecd, 2018-04-06T02:26:57.615Z)
OS version: Windows_NT x64 10.0.16299

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM)2 Quad CPU    Q6600  @ 2.40GHz (4 x 2394)|
|Memory (System)|7.93GB (4.43GB free)|
|Process Argv|C:\Program Files\Microsoft VS Code\Code.exe C:\Users\redacted\Desktop\styles - Copy.xml|
|Screen Reader|no|
|VM|0%|

</details>Extensions: none

**Comment by aeschli** [comment]
Yes, for performance reasons, lines longer than 20k are not indexed.

duplicate of #44206

**Comment by skycommand** [comment]
@aeschli Let me get this straight: For the fear of low performance, VS Code is not performing at all?

If your approach to syntax highlighting is line-based, then you are looking at the whole subject matter the wrong way from the beginning, letting a special character like line break dictate the apps behavior.

Well. It seems I have to go back to Notepad++ after all.
